### PR TITLE
Support for exclude globs at the `git log` level

### PR DESCRIPTION
**Pull request by zricethezav**
### What does this PR do?

Adds a `git` scan mode option to exclude globs at the `git log` level. This is useful if a repo has known files/directories that should be ignored by scans like `/vendor` for Go projects, or `/node_modules ` for node projects.

try it yourself:

```
# This should only report false positives from `go.mod`

git clone -b feature-exclude https://github.com/trufflesecurity/trufflehog.git
cd trufflehog
go build && ./trufflehog  git --no-verification --exclude-globs='**/*.proto,**/*.go' file://$PWD
```

### CODE REVIEW

File: pkg/gitparse/gitparse.go [hunk 1]
```diff
@@ -126,9 +126,13 @@ func (c *Parser) RepoPath(ctx context.Context, source string, head string, abbre
 	} else {
 		args = append(args, "--all")
 	}
+	if len(excludedGlobs) != 0 {
```
Comment (dustin-decker) [inline pkg/gitparse/gitparse.go:129]: Nit: This length check if unnecessary, the inner iteration over the slice will do the same thing.

File: main.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ var (
 	gitScanURI          = gitScan.Arg("uri", "Git repository URL. https://, file://, or ssh:// schema expected.").Required().String()
 	gitScanIncludePaths = gitScan.Flag("include-paths", "Path to file with newline separated regexes for files to include in scan.").Short('i').String()
 	gitScanExcludePaths = gitScan.Flag("exclude-paths", "Path to file with newline separated regexes for files to exclude in scan.").Short('x').String()
```
Comment (dustin-decker) [inline main.go:62]: Do you think that we should deprecate the exclude path regex files longer term? It seems less economic to me than globbing, which is more prevalent among CLI tools that deal with paths and files.

File: main.go [hunk 3]
```diff
@@ -60,6 +60,7 @@ var (
...
+	gitScanExcludeGlobs = gitScan.Flag("exclude-globs", "Comma separated list of globs to exclude in scan.").String()
```
Comment (dustin-decker) [inline main.go:63]: Is it worth noting that this option is more performant than the exclude paths option?

**Review by dustin-decker — APPROVED** [review]
Approved with some questions and a nit

File: pkg/gitparse/gitparse.go (hunk 1 above)
Comment (zricethezav) [reply pkg/gitparse/gitparse.go:129]: whoops, that's a relic. Good catch

File: main.go (hunk 3 above)
Comment (zricethezav) [reply main.go:63]: That's a good call

**Pull request merged**